### ISSUE TITLE
What is corePress?

### ISSUE THREAD

**Issue by danieltj27** [corePress/corePress#7]
I've just learnt of 'corePress'. First we had ClassicPress, then calmPress and now corePress? What's the difference between these three projects and who is 'owning' the project? It seems to be some kind of joint effort between Classic/calm-Press forks, no?

**Comment by stefanos82** [comment]
@danieltj27 The objective of corePress is to have a common `core` ground between calmPress and ClassicPress, and to be completely honest with you with any <foo>Press project wants to initiate either a new fork or a new Press project from scratch.

My idea is to extract all common mechanisms each WordPress-based project has and place them under a module / package format so any user can choose which parts of his / her project should be on by default.

corePress does not compete with any Press project; on the contrary, it tries to become the Composer with packagist for <foo>Press websites.

Now I need a nice repo description and a decent README which I have already asked in #6 .

**Comment by danieltj27** [comment]
So effectively what you're saying is, that it's a plugin & theme repository shared by ClassicPress and calmPress? I'm just curious how this would be useful for people using WordPress plugins? I've read multiple times that WordPress plugins will be supported by both projects so the WordPress.org directories would still benefit each project.

How does fracturing this further help the end user?

**Comment by stefanos82** [comment]
My idea is to support WordPress as well and pretty much any project that uses the same concept as WordPress's back-end code.

What I would like though is us introducing a compatible API that could be used with existing Press projects that "speak" the same common language so they can be interconnected with each other with as little friction as possible.

But for now, the idea is to share a common ground between calmPress and ClassicPress.

In other words, think of this as CorePress-FIG, much like PHP-FIG.

**Comment by stefanos82** [comment]
I have updated the README to clarify the repo's actual nature.

**Comment by markkap** [comment]
> In other words, think of this as CorePress-FIG, much like PHP-FIG.

+1

if we get some common code, and maybe even code that can be used in WordPress plugins, even better, but from my POV, the standardization is the main goal.

**Comment by stefanos82** [comment]
@markkap yep, the standardization is all I have had in mind in the first place.

WordPress plugins and themes could use it if they wanted to, but that's another story...
